src/client/create.ts

```diff
@@ -59,11 +70,34 @@ async function createGame(
     headers,
   };
 
-  await fetch(url, req)
-    .then((res) => {
-      window.location.assign(res.url);
-    })
-    .catch((error) => {
-      throw new Error(`failed to create game: ${error})`);
+  return fetch(url, req).then((res) => {
+    if (res.status === 200) {
+      return res.url;
+    }
+    // We'll expect the body to contain an error
+    // in JSON format.
+    return res.text().then((json) => {
+      throw new Error(json);
     });
+  });
+}
+
+function hideCreation() {
+  lobbyDiv.classList.add("invisible");
+  createForm.setAttribute("disabled", "true");
+}
+
+function showCreation() {
+  lobbyDiv.classList.remove("invisible");
+  createForm.classList.remove("invisible");
+  createForm.setAttribute("disabled", "false");
+}
+
+function validateInput(playerName: string, gameName: string) {
+  if (!isValidName(playerName)) {
+    throw new InvalidName(playerName);
+  }
+  if (!isValidName(gameName)) {
+    throw new InvalidName(playerName);
```

Comment:
should this be `throw new InvalidName(gameName);`?

Reply:
Yes it should!! Thank you >.<